tasks/build/os_shell_scripts.js

```diff
@@ -0,0 +1,41 @@
+import {unlink} from 'fs';
+import {join, extname} from 'path';
+import {promisify} from 'bluebird';
+import {ncp} from 'ncp';
+const pncp = promisify(ncp);
+const punlink = promisify(unlink);
+
+module.exports = function (grunt) {
+  grunt.registerTask('_build:osShellScripts', function () {
+    const done = this.async();
+    const source = 'build/kibana/bin';
+    const platforms = grunt.config.get('platforms');
+
+    Promise.resolve()
+      .then(() => allPlatforms(platform => punlink(join(platform.buildDir, 'bin')))
+      .then(() => allPlatforms(platform => pncp(source, join(platform.buildDir, 'bin'))))
+      .then(() => allPlatforms(removeExtraneousShellScripts)))
+      .then(() => done)
```

Comment:
Since you are doing all of these in sequence and you have a single error handler for everything, I recommend replacing the promise chaining and built in `this.async` behavior with native (transpiled) async/await.